elephant/test/test_spike_train_generation.py

```diff
@@ -69,6 +68,66 @@ def test_threshold_detection(self):
         except AttributeError: # If numpy version too old to have allclose
             self.assertTrue(np.array_equal(spike_train,spike_train))
 
+
+class AnalogSignalPeakDetectionTestCase(unittest.TestCase):
+
+    def setUp(self):
+        curr_dir = os.path.dirname(os.path.realpath(__file__))
+        npz_file_loc = os.path.join(curr_dir, 'spike_extraction_test_data.npz')
+        iom2 = neo.io.PyNNNumpyIO(npz_file_loc)
+        data = iom2.read()
+        self.vm = data[0].segments[0].analogsignals[0]
+        self.true_time_stamps = [0.0124,  0.0354,  0.0713,  0.1192,  0.1695,
+                                 0.2201,  0.2711] * second
+
+    def test_peak_detection_time_stamps(self):
+        # Test with default arguments
+        result = stgen.peak_detection(self.vm)
+        self.assertEqual(len(self.true_time_stamps), len(result))
+        self.assertIsInstance(result, neo.core.SpikeTrain)
+
+        try:
```

Comment:
assert_array_almost_equal asserts the equality of arrays up to a decimal place (6 decimals by default). This is not necessary for integers, which is why self.assertEqual is used to compare the length of arrays. To be honest I took the try: except: construct from the already implemented AnalogSignalThresholdDetectionTestCase and I am not entirely sure what it does and why it catches specifically the AttributeError but I will look into it.